modules/internal/Bytes.chpl

```diff
@@ -86,27 +86,46 @@ module Bytes {
   }
 
   /*
-    Creates a new :type:`bytes` which borrows the internal buffer of a
-    `c_string`. If the buffer is freed before the :type:`bytes` returned
+    Creates a new :type:`bytes` which borrows the memory allocated for a
+    `c_ptr`. If the buffer is freed before the :type:`bytes` returned
     from this function, accessing it is undefined behavior.
 
-    :arg s: `c_string` to borrow the buffer from
+    :arg x: `c_ptr` to borrow as a buffer
+    :type x: `c_ptr(c_uchar)` or `c_ptr(c_char)`
 
-    :arg length: Length of `s`'s buffer, excluding the terminating
-                 null byte.
+    :arg length: Length of `x`, excluding the terminating null byte.
+    :type length: `int`
+
+    :returns: A new :type:`bytes`
+  */
+  inline proc type bytes.createBorrowingBuffer(x: c_ptr(?t),
```

Comment:
I think this is handled by the call to `bytes.createBorrowingBuffer` defined on ln 180, which throws a compiler error if the type of `c_ptr(?t)` is not `byteType` or `c_char`.

```chapel
  inline proc type bytes.createBorrowingBuffer(x: c_ptr(?t), length: int,
                                                  size: int) : bytes {
    if t != byteType && t != c_char {
      compilerError("Cannot create a bytes with a buffer of ", t:string);
    }
    var ret: bytes;
    initWithBorrowedBuffer(ret, x:bufferType, length, size);
    return ret;
  }
```

Reply:
IMO it should work on a platform where `char` is unsigned in C; to do so it would need slightly different logic, the most easy being to check for `t` being `int(8)` or `uint(8)`.

Reply:
sorry, that copy of the code is prior to me updating all those locations to `int(8)` and `uint(8)`

Reply:
Sounds good, in that case